idasen/cli.py

```diff
@@ -153,13 +153,22 @@ async def init(args: argparse.Namespace) -> int:
         print(
             f"Created new configuration file at: {IDASEN_CONFIG_PATH}", file=sys.stderr
         )
+        print("'idasen pair' can be used to pair to desk.")
 
     return 0
 
 
 async def pair(args: argparse.Namespace) -> None:
-    async with IdasenDesk(args.mac_address, exit_on_fail=True) as desk:
-        await desk.pair()
+    try:
+        async with IdasenDesk(args.mac_address, exit_on_fail=True) as desk:
+            await desk.pair()
+    except NotImplementedError:
+        print(
+            "The pair subcommand does not function reliably on macOS.\n"
+            "A pairing dialogue is shown if the OS deems that pairing is necessary.\n"
+            "Retrying can help.\n\n"
+            "See docs at https://bleak.readthedocs.io/en/latest/backends/macos.html"
+        )
```

Comment:
```suggestion
    except NotImplementedError as e:
        if platform.system() == "Darwin":
            print(
                "The pair subcommand does not function reliably on macOS.\n"
                "A pairing dialogue is shown if the OS deems that pairing is necessary.\n"
                "Retrying can help.\n\n"
                "See docs at https://bleak.readthedocs.io/en/latest/backends/macos.html"
            )
            return 1
        else:
            raise e
```

* should return 1 (change type hint to `Optional[int]`) since it did fail.
* might return `NotImplementedError` on other platforms in the future, good to check for MacOS here.